openff/interchange/interop/openmm/_topology.py

```diff
@@ -173,6 +186,21 @@ def to_openmm_topology(
                 order=bond_order,
             )
 
+    if has_virtual_sites and not collate:
+        virtual_site_chain = openmm_topology.addChain(atom_chain_id)
+        virtual_site_residue = openmm_topology.addResidue("VS", virtual_site_chain)
+
+        for molecule_index, molecule in enumerate(topology.molecules):
+            virtual_sites_in_this_molecule: list[VirtualSiteKey] = molecule_virtual_site_map[molecule_index]
```

Comment:
Only briefly looking at things, but it looks to me like that we're really needing is this map. If we could map things to/from the collated form then yes we have to plan for this map being optionally around, but at the same time it's not crazy extra amounts of work to work around it.

Reply:
I'd like to expose a map like this to the user - I have to re-create it a number of times while exporting, might as well cache it on the `Interchange` itself or do something like that - but want to be careful and deliberate when making that change

Reply:
#1076